### DISCUSSION TITLE
Docker Install Help Needed

### DISCUSSION THREAD

**Discussion by KleeziE** [requarks/wiki#6964] [Help / Questions] [answered]
Host: RHEL 8.9
wiki.js version: 2.5
External DB: postgresql 9.6

So running into the following error when trying to run the container with this command-> podman run -d -p 8080:3000 --name wiki --restart unless-stopped -v ./config.yml:/wiki/config.yml:Z redacted.local:5000/requarks/wiki:2.5

```
Loading configuration from /wiki/config.yml... OK
2024-02-01T17:42:42.171Z [MASTER] info: =======================================
2024-02-01T17:42:42.173Z [MASTER] info: = Wiki.js 2.5.300 =====================
2024-02-01T17:42:42.173Z [MASTER] info: =======================================
2024-02-01T17:42:42.173Z [MASTER] info: Initializing...
2024-02-01T17:42:42.643Z [MASTER] info: Using database driver pg for postgres [ OK ]
2024-02-01T17:42:42.656Z [MASTER] info: Connecting to database...
/wiki/node_modules/pg/lib/query.js:143
          throw err
          ^

TypeError: Cannot read properties of null (reading '1')
    at Query.<anonymous> (/wiki/node_modules/knex/lib/dialects/postgres/index.js:148:69)
    at Query.handleReadyForQuery (/wiki/node_modules/pg/lib/query.js:139:14)
    at Client._handleReadyForQuery (/wiki/node_modules/pg/lib/client.js:301:19)
    at Connection.emit (node:events:514:28)
    at /wiki/node_modules/pg/lib/connection.js:119:12
    at Parser.parse (/wiki/node_modules/pg-protocol/dist/parser.js:40:17)
    at Socket.<anonymous> (/wiki/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:514:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)

Node.js v18.17.0
```

config.yml is as follows:
```
port: 3000
bindIP: 0.0.0.0
db:
  type: postgres
  host: redacted.local
  port: 5444
  user: redacted
  pass: redacted
  db: Wiki
  ssl: 0
ssl:
  enabled: 1
  port: 3443
  provider: letsencrypt
  domain: wiki.redacted.com
  subscriberEmail: redacted
logLevel: debug
logFormat: default
ha: 0

```
I have also copied and used the suggested sample config.yml file but got the same result fyi.

Tested DB connection from host w/ python script:

```
#!/usr/bin/python2.4
#
import psycopg2
try:
    db = psycopg2.connect("dbname='Wiki' user='redacted' host='redacted.local' port='5444' password='redacted'")
except:
    exit(1)
 
exit(0)
```

```
sudo python3.11 psqltest.py && echo 'OK' || echo 'FAIL'
OK
```

Tested that the container can reach the db server via:

```
podman exec wiki ping redacted.local
PING redacted.local (10.xx.xx.xx): 56 data bytes
64 bytes from 10.xx.xx.xx: seq=0 ttl=60 time=0.671 ms
```

Any advice is greatly appreciated as we are looking to migrate all of our confluence stuff over to wiki.js.

**Comment by alexgQQ** [comment]
Hello 👋 odd issue,  I'm not very familiar with podman but does it handle networking in a similar way as docker?
I'd check that TCP is good with `podman exec wiki telnet redacted.local 5444` and if that works then I'd try to connect with `psql` through the wiki container with your redacted user for sanity.

Is there anything already in that `Wiki` database?

Past that I'm not too sure though, the minimum postgres version supported is 9.5 and yours isn't far off. Maybe there's an issue with 9.6 we don't know about 🤷

**Comment by fnafox** [comment]
it very much smells like a database issue to me, i think it's worth checking whether your db was created from a non default template.
it might be something not explicitly documented as a requirement such as the character encoding or collation

**Comment by KleeziE** [comment]
DB is using EDB Postgresql 9.6... makes me wonder the same thing. Not sure what to communicate to the DBA for the sake of wiki.js to work, any clues where to find that in the setup script of wiki.js? 

This is a clean/empty db as of last week fyi.

**Reply by fnafox** [reply, marked as answer]
I believe this issue upstream is relevant, it's not the same exact problem, but EDB returns a different value for `select version();` which doesn't get picked up by the regex. I'm not really sure how this could be resolved.
https://github.com/knex/knex/issues/1477#issuecomment-668965455

**Comment by KleeziE** [comment]
Thank you very much, this was the issue. Regex came back as EDB versus Postgresql. Switched to our SQL cluster and it worked fine.

Might be something the wiki.js team could add cause from what the DBAs told me, EDB and Postgresql are essentially identical.
